### PR TITLE
[SYCLomatic] Add 24 thrust APIs for --query-api-mapping option

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/examples/DPCT/Thrust/thrust$$set_union_by_key.cu [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#include <thrust/device_malloc.h>
```
Comment (zhimingwang36) [inline clang/examples/DPCT/Thrust/thrust$$set_union_by_key.cu:1]: Prelease add lit test to see what is the migration result of set_union_by_key

Comment (tomflinda) [reply clang/examples/DPCT/Thrust/thrust$$set_union_by_key.cu:1]: Here is the lit test: https://github.com/oneapi-src/SYCLomatic/blob/73874101f128eaecd0a14f5f52ddf9a1b2723a9f/clang/test/dpct/thrust-algo-part3.cu#L245

Comment (zhimingwang36) [reply clang/examples/DPCT/Thrust/thrust$$set_union_by_key.cu:1]: ref to https://github.com/oneapi-src/SYCLomatic/pull/1195/files#diff-a1b8a2634fb9afbfb3d5eadc1a79f4dc6af2aa8717e72c20dc42807479428ad5

File: clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu [hunk 2]
```diff
@@ -0,0 +1,262 @@
+// UNSUPPORTED: cuda-8.0
```
Comment (tangjj11) [inline clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu:2]: seem like only 23 API lit test

**Review by tangjj11 — APPROVED** [review]

File: clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu (hunk 2 above)
Comment (tomflinda) [reply clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu:2]: lit test for stable_partition is missing, will be added in the updated PR.

File: clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+// UNSUPPORTED: v8.0
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=thrust::set_union_by_key | FileCheck %s -check-prefix=set_union_by_key
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu]: Please add -stop-on-parse-err option for the lit case.

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu (hunk 3 above)
Comment (tomflinda) [reply clang/test/dpct/query_api_mapping/Thrust/thrust_api_test.cu]: okay

File: clang/examples/DPCT/Thrust/thrust$$all_of.cu [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+void test() {
+
+  bool A[3] = {true, true, false};
+  bool result;
+  thrust::host_vector<bool> h_A(A, A + 3);
+  thrust::device_vector<bool> d_A(A, A + 3);
+
+  // Start
+  thrust::all_of(thrust::host /*const thrust::detail::execution_policy_base<
```
Comment (zhimingwang36) [inline clang/examples/DPCT/Thrust/thrust$$all_of.cu]:
```suggestion
/* 1 */  thrust::all_of(thrust::host /*const thrust::detail::execution_policy_base<
```

**Review by tangjj11 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**